service/dap/server.go

```diff
@@ -550,58 +550,66 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		s.config.Debugger.WorkingDir = wdParsed
 	}
 
-	noDebug := false
-	if noDebugArg, ok := request.Arguments["noDebug"]; ok {
-		if v, ok := noDebugArg.(bool); ok && v { // noDebug == true
-			noDebug = true
-		}
-	}
-
-	var err error
-	if noDebug {
-		if s.noDebugProcess, err = s.launchWithoutDebug(program, targetArgs, s.config.Debugger.WorkingDir); err != nil {
+	if noDebug, _ := request.Arguments["noDebug"].(bool); noDebug {
+		cmd, err := s.startNoDebugProcess(program, targetArgs, s.config.Debugger.WorkingDir)
+		if err != nil {
 			s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", err.Error())
 			return
 		}
-	} else {
-		if s.debugger, err = debugger.New(&s.config.Debugger, s.config.ProcessArgs); err != nil {
-			s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", err.Error())
-			return
+		// Skip 'initialized' event.
```

Comment:
I think it would be good to document here that by doing so we avoid debug requests.

Reply:
Done.